jmx-metrics/src/main/resources/target-systems/kafka.groovy

```diff
@@ -14,81 +14,178 @@
  * limitations under the License.
  */
 
-def messagesInPerSec = otel.mbean("kafka.server:type=BrokerTopicMetrics,name=MessagesInPerSec")
-otel.instrument(messagesInPerSec, "kafka.messages.in", "number of messages in per second",
-        "1", "Count", otel.&longValueCallback)
+def messagesInPerSec = otel.mbean("kafka.server:type=BrokerTopicMetrics,name=MessagesInPerSec,topic=*")
+otel.instrument(messagesInPerSec,
+  "kafka.message.count",
+  "The number of messages received by the broker",
+  "messages",
+  [
+    "topic" : { mbean -> mbean.name().getKeyProperty("topic") },
```

Comment:
Are there potential cardinality concerns here? I think very large deployments can have thousands of topics.

Reply:
That's a valid concern. Probably better to just monitor the total messages.